src/CommunityToolkit.Aspire.Keycloak.Extensions/KeycloakPostgresExtension.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Aspire.Hosting;
+using Aspire.Hosting.ApplicationModel;
+
+namespace CommunityToolkit.Aspire.Keycloak.Extensions;
+
+/// <summary>
+/// Provides extension methods for integrating Keycloak resources with PostgreSQL.
+/// </summary>
+public static class KeycloakPostgresExtension
+{
+    /// <summary>
+    /// Configures a Keycloak resource to use a PostgreSQL database.
+    /// Sets up the necessary environment variables for Keycloak to connect to the specified PostgreSQL database.
+    /// </summary>
+    /// <param name="builder">
+    /// The resource builder for the Keycloak resource to be configured.
+    /// </param>
+    /// <param name="database">
+    /// The resource builder for the PostgreSQL database resource that the Keycloak resource will connect to.
+    /// </param>
+    /// <param name="username">
+    /// (Optional) The resource builder for the database username parameter.
+    /// </param>
+    /// <param name="password">
+    /// (Optional) The resource builder for the database password parameter.
+    /// </param>
+    /// <param name="xaEnabled">
+    /// Indicates whether XA transactions are enabled for the database connection.
+    /// </param>
+    public static IResourceBuilder<KeycloakResource> WithPostgres(
+        this IResourceBuilder<KeycloakResource> builder,
+        IResourceBuilder<PostgresDatabaseResource> database, IResourceBuilder<ParameterResource> username,
+        IResourceBuilder<ParameterResource> password,
+        bool xaEnabled = false)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(database);
+
+        EndpointReference ep = database.Resource.Parent.GetEndpoint("tcp");
+
+        string dbName = database.Resource.Name;
+
+        ReferenceExpression jdbcUrl = ReferenceExpression.Create(
+            $"jdbc:postgresql://{ep.Property(EndpointProperty.Host)}:" +
+            $"{ep.Property(EndpointProperty.Port)}/{dbName}");
+
+        builder.WithEnvironment("KC_DB", "postgres")
+            .WithEnvironment("KC_DB_URL", jdbcUrl);
+
+        builder.WithEnvironment("KC_DB_USERNAME", username.Resource);
+        builder.WithEnvironment("KC_DB_PASSWORD", password.Resource);
```

Comment:
Can you get these from the database.Parent instead? This overload is about using an existing postgres right?